hs_rscript_resource/models.py

```diff
@@ -0,0 +1,135 @@
+from django.db import models
+from django.contrib.contenttypes import generic
+
+from mezzanine.pages.page_processors import processor_for
+
+from hs_core.models import BaseResource, ResourceManager, resource_processor, CoreMetaData, AbstractMetaDataElement
+
+from lxml import etree
+
+
+class RScriptResource(BaseResource):
+    objects = ResourceManager('RScriptResource')
+
+    class Meta:
+        proxy = True
+        verbose_name = 'R Script Resource'
+
+    @classmethod
+    def get_supported_upload_file_types(cls):
+        # one file type is supported
+        return ".r"
+
+    @classmethod
+    def can_have_multiple_files(cls):
+        # can have multiple files
+        return True
+
+    @property
+    def metadata(self):
+        md = RScriptMetaData()
+        return self._get_metadata(md)
+
+processor_for(RScriptResource)(resource_processor)
+
+
+class RSMetadata(AbstractMetaDataElement):
+    term = "RSMetadata"
+
+    # program language
+    scriptLanguage = models.CharField(verbose_name='Programming Language', blank=True, max_length=100, default='R',
```

Comment:
@ironfroggy @danames Yeah, Dr. Ames told me that we would likely generalize
this to be a Script Resource type, yet he said to go ahead and make it
applicable only to R Scripts for now but to generalize the code (variable
names, layout, etc) to make it a more universal Script Resource in the
future. I don't know if there is any reason not to just generalize it
now... Thoughts? Dr. Ames?

On Mon, Jan 4, 2016 at 11:58 AM, ironfroggy <notifications@github.com>
wrote:

> In hs_rscript_resource/models.py
> <https://github.com/hydroshare/hydroshare/pull/853#discussion_r48765537>:
>
> > +        # can have multiple files
> > +        return True
> > +
> > +    @property
> > +    def metadata(self):
> > +        md = RScriptMetaData()
> > +        return self._get_metadata(md)
> > +
> > +processor_for(RScriptResource)(resource_processor)
> > +
> > +
> > +class RSMetadata(AbstractMetaDataElement):
> > +    term = "RSMetadata"
> > +
> > +    # program language
> > +    scriptLanguage = models.CharField(verbose_name='Programming Language', blank=True, max_length=100, default='R',
>
> FWIW, I think a single Script Resource type with configurable languages
> makes a lot more sense than a separate type for each language.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hydroshare/hydroshare/pull/853/files#r48765537>.
>